front/lib/api/assistant/conversation.ts

```diff
@@ -620,6 +620,29 @@ export async function* postUserMessage(
       })
     )
   );
+
+  async function createOrUpdateParticipation() {
+    if (user) {
+      const participant = await ConversationParticipant.findOne({
+        where: {
+          conversationId: conversation.id,
+          userId: user.id,
+        },
+      });
+      if (participant) {
+        return participant.update({
+          action: "posted",
+        });
+      } else {
+        return ConversationParticipant.create({
+          conversationId: conversation.id,
+          userId: user.id,
+          action: "posted",
+        });
+      }
+    }
+  }
+  await createOrUpdateParticipation();
```

Comment:
You add a sequential call to the db on the critical path here.

Can we parallelize with something else?

Reply:
I think we are talking about less than 1ms or 2ms at worst. Do you think it's worst doing this optimization?

Reply:
Adding a commit for //